cl/users/tests.py

```diff
@@ -905,3 +909,646 @@ def test_update_ban_object(self) -> None:
         # Checks email ban is updated with the hard bounce subtype
         self.assertEqual(email_ban.count(), 1)
         self.assertEqual(email_ban[0].event_sub_type, SUB_TYPES.GENERAL)
+
+
+class CustomBackendEmailTest(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.user = UserFactory()
+        test_dir = Path(settings.INSTALL_ROOT) / "cl" / "users" / "test_assets"
+        with open(test_dir / "hard_bounce.json", encoding="utf-8") as file:
+            cls.hard_bounce_asset = json.load(file)
+        with open(
+            test_dir / "general_soft_bounce.json", encoding="utf-8"
+        ) as file:
+            cls.soft_bounce_asset = json.load(file)
+        with open(
+            test_dir / "msg_large_bounce.json", encoding="utf-8"
+        ) as file:
+            cls.soft_bounce_msg_large_asset = json.load(file)
+
+        cls.attachment_150 = test_dir / "file_sample_150kB.pdf"
+        cls.attachment_500 = test_dir / "file_example_500kB.pdf"
+
+    def send_signal(self, test_asset, event_name, signal) -> None:
+        """Function to dispatch signal that mocks a SNS notification event
+        :param test_asset: the json object that contains notification
+        :param event_name: the signal event name
+        :param signal: the signal corresponding to the event
+        :return: None
+        """
+        # Prepare parameters
+        raw = json.dumps(test_asset)
+        notification = json.loads(raw)
+        message = json.loads(notification["Message"])
+        mail_obj = message.get("mail")
+        event_obj = message.get(event_name, {})
+
+        # Send signal
+        signal_kwargs = dict(
+            sender=self,
+            mail_obj=mail_obj,
+            raw_message=raw,
+        )
+        signal_kwargs[f"{event_name}_obj"] = event_obj
+        signal.send(**signal_kwargs)
+
+    def get_email_body(self, message):
+        """Function to retrieve text/html and text/plain body of an email
+        :param message: the message to extract the body content
+        :return: plaintext_body and html_body strings
+        """
+        plaintext_body = ""
+        html_body = ""
+        for part in message.walk():
+            if part.get_content_type() == "text/plain":
+                plaintext_body = part.get_payload()
+                if html_body:
+                    break
+            if part.get_content_type() == "text/html":
+                html_body = part.get_payload()
+                if plaintext_body:
+                    break
+
+        return plaintext_body, html_body
```

Comment:
Not better to use the other method like this one that you have in the code itself?

Reply:
Of course, changed!